### ISSUE TITLE
Feature Request: Better Tagging for Slaves

### ISSUE THREAD

**Issue by JoshuaTPritchett** [rht-labs/labs-ci-cd#129]
#### Feature Request

As a developer, I would like more control over which Jenkins Slaves deploy.

#### Proposed Solution

Add in extra tags under each slave as: jenkins-slave-<slave_name>

**Comment by sherl0cks** [comment]
Deploy to what? If you mean deploy and run as part of a slave in a pipeline, that is controlled by the pipeline itself

**Comment by JoshuaTPritchett** [comment]
When I run the inventory with casl, it would be nice to control which slave deploys for testing purposes. Instead of deploying all of them.

**Comment by sherl0cks** [comment]
Running the full inventory will build all the slaves, but it will not directly deploy a pod running the image. Jenkins is responsible for running the pods as slaves using the syntax I shared above. So the java slave today is spun up, but that's the result of Jenkins pipeline running. You can not deploy that build config or even not deploy Jenkins and the slave won't deploy.

If you'd like to limit the slaves that are built, you can use filter tags to deploy a subset of the inventory.

**Comment by JoshuaTPritchett** [comment]
What I am thinking and I am wrong, is that right now when deploying all of the components via casl-ansible an example base command would be something along these lines:
```
ansible-playbook -i inventory/ roles/casl-ansible/playbooks/openshift-cluster-seed.yml --extra-vars="filter_tags=jenkins,jenkins-slaves,projects"
```
Instead of deploying all slaves I was thinking that having an extra tag under each slave like the ansible slave would allow for redeployment like so:
```
ansible-playbook -i inventory/ roles/casl-ansible/playbooks/openshift-cluster-seed.yml --extra-vars="filter_tags=ansible-slave, zap-slave"
```
That way when testing you can tag specific slaves and potentially redeploy via casl as needed.

**Comment by sherl0cks** [comment]
Ahh ok. Language issue here. You're talking about tags in the inventory. Generally, speaking the language we us in this case is "apply parts of the inventory", since under the covers the oc apply command is used. Deploy generally has a connotation of an app running.

That said, I support tags to be add to the invetory for this use case. I honestly thought we already had them.

**Comment by JoshuaTPritchett** [comment]
Ah, totally makes sense. I am completely self-taught, sorry if my language is a bit off :-p Is this approved?

**Comment by sherl0cks** [comment]
No prob. Language is a pain. 

Yes I support a PR that provides this feature

**Comment by JoshuaTPritchett** [comment]
Another quick question, it seems that within the `all.yml` inventory the ansible slave is being applied twice. Was this intended or a dup?

**Comment by sherl0cks** [comment]
Appears to be a bad merge. Good catch. The nature of the applier being idempotent means we wont get two copies of the slave, but it is confusing. Let's clean that up with the tags.

**Comment by sherl0cks** [comment]
resolved by #130
